src/test/java/edu/ucla/library/bucketeer/ImageUploadIT.java

```diff
@@ -0,0 +1,192 @@
+ * From the ticket that requested this integration test (IIIF-166):
+ * This should confirm that a request at the front end of the service goes
+ * through the router, to the converter, and ends up in the S3 upload (with the
+ * item being requested being uploaded (assuming it exists on our file system)).
+ * We won't test callback at this point in time. That's for a later ticket.
+ */
+public class ImageUploadIT {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImageUploadIT.class, Constants.MESSAGES);
+    private static final Integer PORT = Integer.parseInt(System.getProperty("http.port"));
+    private static final String DEFAULT_ACCESS_KEY = "YOUR_ACCESS_KEY";
+    private static final String DEFAULT_SECRET_KEY = "YOUR_SECRET_KEY";
+    private static final String DEFAULT_S3_BUCKET = "cantaloupe-jp2k";
+    private static final String TEST_FILE_PATH = "src/test/resources/images/test.tif";
+    private static final String SLASH = "/";
+    private static final String PING = "/ping";
+    private static final String UTF8 = "UTF-8";
+    private static final String DOESNOTEXIST = ") does not exist!";
+    private static final String HELLO = "Hello";
+
+    private static String myS3Bucket;
+    private static String myS3AccessKey;
+    private static String myS3SecretKey;
+    private static AWSCredentials myAWSCredentials;
+
+    private static Vertx vertx = Vertx.vertx();
+
+    /** We can't, as of yet, execute these tests without a non-default S3 configuration */
+    private static boolean isExecutable;
+
+    private File myTIFF;
+    private String myUUID;
+    private String myDerivativeJP2;
+    private String myImageLoadRequest;
+    private AmazonS3 myAmazonS3;
+    private int myStatusCode;
+
+    /**
+     * Use RestAssured to connect to our service, and fetch our configuration
+     */
+    @BeforeClass
+    public static void configureRestAssured() {
+        RestAssured.baseURI = "0.0.0.0";
+        RestAssured.port = PORT;
+        LOGGER.debug(MessageCodes.BUCKETEER_021, RestAssured.port);
+
+
+        final ConfigRetriever configRetriever = ConfigRetriever.create(vertx);
+
+        configRetriever.getConfig(config -> {
+            if (config.succeeded()) {
+                LOGGER.debug(MessageCodes.BUCKETEER_038);
+                final JsonObject jsonConfig = config.result();
+                myS3Bucket = jsonConfig.getString(Config.S3_BUCKET, DEFAULT_S3_BUCKET);
+                myS3AccessKey = jsonConfig.getString(Config.S3_ACCESS_KEY, DEFAULT_ACCESS_KEY);
+                myS3SecretKey = jsonConfig.getString(Config.S3_SECRET_KEY, DEFAULT_SECRET_KEY);
+
+                // We need to determine if we'll be able to run the S3 integration tests so we can skip if needed
+                if (jsonConfig.containsKey(Config.S3_ACCESS_KEY) && !jsonConfig.getString(Config.S3_ACCESS_KEY,
+                        DEFAULT_ACCESS_KEY).equalsIgnoreCase(DEFAULT_ACCESS_KEY)) {
+                    isExecutable = true;
```

Comment:
Without waiting for this lambda function to complete you could end up checking isExecutable before it's been set. You need to not complete the configureRestAssured function until the lambda has been completed.

Reply:
You can borrow the async.complete() stuff from the test where you found this config checking code and that should get it working.